From 5a786ac6ad27f6651348126da66ae318a2040399 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sat, 19 Jul 2014 04:30:17 -0400
Subject: [pci] msi: Fix leak of msi_attrs

Message-id: <20140719043017.19438.46380.stgit@gir.stowe>
Patchwork-id: 86034
O-Subject: [RHEL7.1 PATCH 72/81] PCI/MSI: Fix leak of msi_attrs
Bugzilla: 1110895
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110895
Upstream Status: 29237756477d5e777e39257d94af449fc66026c2

commit 29237756477d5e777e39257d94af449fc66026c2
Author: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date:   Thu Feb 13 10:47:35 2014 -0700

    PCI/MSI: Fix leak of msi_attrs

    Coverity reported that I forgot to clean up some allocated memory on the
    error path in populate_msi_sysfs(), so this patch fixes that.

    Thanks to Dave Jones for pointing out where the error was, I obviously
    can't read code this morning...

    Found by Coverity (CID 1163317).

    Fixes: 1c51b50c2995 ("PCI/MSI: Export MSI mode using attributes, not kobject
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
    Cc: Dave Jones <davej@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/msi.c b/drivers/pci/msi.c
index 6c423f9..cfce591 100644
--- a/drivers/pci/msi.c
+++ b/drivers/pci/msi.c
@@ -617,6 +617,7 @@ error_attrs:
   ++count;
   msi_attr = msi_attrs[count];
  }
+ kfree(msi_attrs);
  return ret;
 }
 
-- 
1.7.1